pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -163,6 +164,12 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environme
 	for _, service := range push.Services(proxy) {
 		destRule := push.DestinationRule(proxy, service)
 		for _, port := range service.Ports {
+			destServiceMTLSMode := authn_v1alpha1_applier.MTLSUnknown
+			// For now, require enableAutoMtls mode set (to true) explicitly to recorgnize this mode.
+			if env.Mesh.EnableAutoMtls != nil && env.Mesh.EnableAutoMtls.GetValue() {
+				destServiceMTLSMode = authn_v1alpha1_applier.GetMutualTLSMode(
+					push.AuthenticationPolicyForWorkload(service, port))
+			}
```

Comment:
this is inverted. We need to only check if the destination service has strict mode, irrespective of autoMTLS. If it has strict mode, then we just do mtls.

Reply:
We don't want to change the existing behavior. If doing the other way, then users don't have way to opt out of this new feature.